### PR TITLE
Prepare dotty-dokka to document itelf

### PR DESCRIPTION
**Pull request by romanowski**
Add action to publish documentation from each build on master to gh-pages
Add support for rendering index documents

Add seleton documentation

After this PR is merged [our documentation stab](https://contextbuddy.s3.eu-central-1.amazonaws.com/dokka-dotty/pr-97/self/index.html) will be publish to our gh-pages branch after each build on `master`

We are also generating documentation for dotty quite nicely [see here](https://contextbuddy.s3.eu-central-1.amazonaws.com/dokka-dotty/pr-97/stdLib/index.html)

### CODE REVIEW

File: .github/workflows/CI.yaml [hunk 1]
```diff
@@ -51,4 +51,15 @@ jobs:
               aws s3 rm $dest
               aws s3 sync output $dest
 
+      - name: Update gh-pages
+        run: |
+             if [ $GITHUB_REF = "refs/heads/master" ]; then
+                git config --global user.email "$(git log -1 --format=%ae)"
+                git config --global user.name "$(git log -1 --format=%an)"
+                git subtree add --prefix=our-site origin gh-pages
+                cp -r  output/self/* our-site
```
Comment (abgruszecki) [inline .github/workflows/CI.yaml:61]: Shouldn't we first remove all files in `our-site`, to get rid of outdated files?

File: build.sbt [hunk 2]
```diff
@@ -43,13 +43,13 @@ buildDokkaApi := {
 
 val generateSelfDocumentation = inputKey[Unit]("Generate example documentation")
 generateSelfDocumentation := {
-  run.in(Compile).fullInput(" -o output/self -t target/scala-0.26/classes").evaluated // TODO #35 proper sbt integration
+  run.in(Compile).fullInput(" -o output/self -t target/scala-0.26/classes -d documentaiton").evaluated // TODO #35 proper sbt integration
```
Comment (abgruszecki) [inline build.sbt]: Correct to "documentation"?

**Review by abgruszecki — APPROVED** [review]
Otherwise LGTM, looks good!

File: .github/workflows/CI.yaml (hunk 1 above)
Comment (romanowski) [reply .github/workflows/CI.yaml:61]: good point, I will do the followup

**Comment by romanowski** [comment]
I've fixed both reported problems.

**Pull request merged**